docs/ibc/upgrades/quick-guide.md

```diff
@@ -0,0 +1,44 @@
+# How to Upgrade IBC Chains and their Clients
+
+The information in this doc for upgrading chains is relevant to SDK chains. However, the guide for counterparty clients is relevant to any Tendermint client that enables upgrades.
+
+## Upgrading Chains
+
+
+### IBC Client Breaking Upgrades
+
+IBC-connected chains must perform an IBC upgrade if their upgrade will break counterparty IBC clients. The current IBC protocol supports upgrading tendermint chains for a specific subset of IBC-client-breaking upgrades. Here is the exhaustive list of IBC client-breaking upgrades and whether the IBC protocol currently supports such upgrades.
+
+Note: Since upgrades are only implemented for Tendermint clients, this doc only discusses upgrades on Tendermint chains that would break counterparty IBC Tendermint Clients.
+
+1. Changing the Chain-ID: **Supported**
+2. Changing the UnbondingPeriod: **Partially Supported**, chains may increase the unbonding period with no issues. However, decreasing the unbonding period may irreversibly break some counterparty clients. Thus, it is **not recommended** that chains reduce the unbonding period.
+3. Changing the height (resetting to 0): **Supported**, so long as chains remember to increment the revision number in their chain-id.
+4. Changing the ProofSpecs: **Supported**, this should be changed if the proof structure needed to verify IBC proofs is changed across the upgrade. Ex: Switching from an IAVL store, to a SimpleTree Store
+5. Changing the UpgradePath: **Supported**, this might involve changing the key under which upgraded clients and consensus states are stored in the upgrade store, or even migrating the upgrade store itself.
+6. Migrating the IBC store: **Unsupported**, as the IBC store location is negotiated by the connection.
+7. Upgrading to a IBC path-breaking version of IBC: **Unsupported**, as IBC version is negotiated on connection handshake.
```

Comment:
Isn't it supported so long as the old versions use the old path? Adding support for a new version with new path should be supported

Reply:
It isn't supported through the upgrade module.

Reply:
Consider chainA and chainB are connected by an IBC channel. chainA decides to upgrade to a different version of IBC such that the paths in `24-host` have changed. The client on `chainB` needs to upgrade its `Verification` methods to use the new paths.

This is theoretically possible by using the upgraded client state and then "converting" it into a new type that implements the different `Verification` method.

However, the IBC version that is used is negotiated in the connection. So that field needs to change as well which isn't supported with client upgrades.

Upgrading and breaking IBC paths is theoretically possible, it just isn't implemented yet and client upgrades alone will not upgrade the connection.

Technically the new IBC version could also store information in the old paths as well, and thus create "backwards-compatibility" for old connections, but this wouldn't be path breaking.

cc: @cwgoes

Reply:
I'd prefer to avoid duplicating storage, though it is possible. Generally upgrading the version of *IBC* is a completely separate concern from upgrading a *particular* light client algorithm, and it should be treated as such - for example, the IBC version needs to be agreed on by both ends, while the light client versions are independent.

Reply:
(sorry, I may have misunderstood the original comment - the light client upgrades should support changing paths used *by the light client algorithm*, but not IBC protocol paths in general, I think)

Reply:
Maybe we could add this for clarity. It may not be immediately clear what a `IBC path-breaking version of IBC` is. But it's really just a non-backwards compatible version of IBC (IBC 2.0) ie we cannot upgrade IBC connection or channel versions with this upgrade mechanism